### PR TITLE
Trunk improvements

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

**Comment by abelsiqueira** [comment]
Missing `Logging` on `REQUIRE`'.
Apart from that, it's very nice to not have exceptions thrown anymore.

**Comment by dpo** [comment]
Oops. That wasn't supposed to be there. I've been breaking up a list of changes into small PRs but that slipped through the cracks.

**Comment by dpo** [comment]
Nevermind. I need to submit a PR that introduces logging before this. Separating it out didn't quite work.

Another issue is that `@.` isn't supported in julia 0.5.

**Comment by abelsiqueira** [comment]
We could make a release for Optimize supporting 0.5 and drop support.

**Comment by dpo** [comment]
Failures across the board. Seems to be something with environment variables not being set properly...

File: src/trust-region/trust-region.jl [hunk 1]
```diff
@@ -24,18 +24,17 @@ and the following function:
 """
 @compat abstract type AbstractTrustRegion end
 
-"""`ρ = ratio(nlp, f, f_trial, Δm, x_trial, step, slope)`
+"""`ared, pred = aredpred(nlp, f, f_trial, Δm, x_trial, step, slope)`
```
Comment (abelsiqueira) [inline src/trust-region/trust-region.jl]: `ratio(nlp...)` is used by TRON, so tron is failing (line 41 of tron-trust-region.jl).

File: src/solver/trunk.jl [hunk 2]
```diff
@@ -123,9 +122,21 @@ function trunk(nlp :: AbstractNLPModel;
...
+      ared, pred = aredpred(nlp, f, ft, Δq, xt, s, slope)
+      if pred ≥ 0
+        status = "nonnegative predicted reduction"
+        stalled = true
+        continue
+      end
+      tr.ratio = ared / pred
       if !monotone
-        ρ_hist = ratio(nlp, fref, ft, σref + Δq, xt, s, slope)
+        ared_hist, pred_hist = ratio(nlp, fref, ft, σref + Δq, xt, s, slope)
```
Comment (abelsiqueira) [inline src/solver/trunk.jl]: ratio -> aredpred

File: src/solver/trunk.jl [hunk 3]
```diff
@@ -87,13 +82,13 @@ function trunk(nlp :: AbstractNLPModel;
...
+    ared, pred = aredpred(nlp, f, ft, Δq, xt, s, slope)
+    if pred ≥ 0
+      status = "nonnegative predicted reduction"
       stalled = true
       continue
     end
+    tr.ratio = ared / pred
 
     if !monotone
       ρ_hist = ratio(nlp, fref, ft, σref + Δq, xt, s, slope)
```
Comment (abelsiqueira) [inline src/solver/trunk.jl]: ratio->aredpred

**Comment by abelsiqueira** [comment]
It works after the changes on `tron`, so I'm merging. Thanks.

**Pull request merged**